x-pack/plugins/canvas/public/components/embeddable_flyout/flyout.tsx

```diff
@@ -74,11 +75,17 @@ export const AddEmbeddablePanel: React.FunctionComponent<FlyoutProps> = ({
   );
 
   const onSelect = useCallback(
-    (id: string, type: string) => {
+    (id: string, type: string, isByValueEnabled?: boolean): void => {
       const partialElement = {
         expression: `markdown "Could not find embeddable for type ${type}" | render`,
       };
-      if (allowedEmbeddables[type]) {
+
+      if (isByValueEnabled) {
```

Comment:
this by value flag is used even for by reference embeddables?

Reply:
Yeah, my intention is to favor the new generic embeddable as the default expression when adding an embeddable from the library.

```
if (isByValueEnabled) {
        const config = encode({ id });
        partialElement.expression = `embeddable config="${config}" 
  type="${type}" 
| render`;
      }
```
We're using the new `embeddable` expression introduced in this PR if the by-value lab is enabled.

If the lab is disabled, we use the original logic which builds an expression based on embeddable type from our mapping of allowed embeddables.

Reply:
Gotcha. I wonder if it makes sense to rename the lab flag to something that isn't necessarily by-value specific. When I'm reading this code and I see a flag checking for by-value to be enabled, my brain makes the assumption that the code I'm going to be reading next has to do with by-value embeddables which isn't necessarily the case. If it's going to be a pain to re-name the flag, maybe a comment explaining that by-reference embeddables are still handled in the block -- it's just going to use the new function